pil-analyzer/src/statement_processor.rs

```diff
@@ -523,41 +505,13 @@ where
             name: name.to_string(),
             polynomial,
             array_index,
-            index: self
-                .evaluate_expression_to_int(index)
+            index: untyped_evaluator::evaluate_expression_to_int(self.driver, index)
                 .unwrap()
                 .try_into()
                 .unwrap(),
         })]
     }
 
-    /// Turns a Type<Expression> to a Type<u64> by evaluating the array length expressions.
```

Comment:
I think this will go away once the basic patterns are merged and the merge target changes to "main".